fs/readdircount_linux_amd64.go

```diff
@@ -0,0 +1,62 @@
+// +build linux,amd64
+
+package fs
+
+import (
+	"fmt"
+	"os"
+	"unsafe"
+
+	"syscall"
+)
+
+func countDirEntries(buf []byte, n int) int {
+	count := 0
+	buf = buf[:n]
+	for len(buf) > 0 {
+		// see man page getdents(2) for struct linux_dirent64
+		reclenOffset := unsafe.Offsetof(syscall.Dirent{}.Reclen)
+		reclen := *(*uint16)(unsafe.Pointer(&buf[reclenOffset]))
+
+		inoOffset := unsafe.Offsetof(syscall.Dirent{}.Ino)
+		ino := *(*uint64)(unsafe.Pointer(&buf[inoOffset]))
```

Comment:
There is no need to use unsafe pointers , see https://github.com/golang/go/blob/9ed0715bb66dbbd0f597f93d0bc70a3d769b1b10/src/syscall/dirent.go#L50

I wonder if we also need the bounds hint